### ISSUE TITLE
`dj.U` under restriction returns relation with no primary key

### ISSUE THREAD

**Issue by eywalker** [datajoint/datajoint-python#373]
When I use `dj.U('attr')` and restrict it with another relation in which `attr` is a non-primary key attribute, then the resulting relation (containing all unique values of `attr` found in the relation) has no primary key.

Here is an error reproducing code snippet:

With the following table definition,
```python
@schema
class Test(dj.Lookup):
    definition = """
    id: int
    ---
    types: varchar(128)
    """
    contents = enumerate(['a', 'b', 'c', 'c', 'a', 'd', 'a', 'a'])
```

The following results:
```
>>> dj.U('types') & Test()
types    
+-------+
a        
b        
c        
d        
 (4 tuples)

>>> (dj.U('types') & Test()).primary_key
[]
```

**Comment by dimitri-yatsenko** [comment]
The correct syntax would be `dj.U('types') & Test()`

Your examples should have raised an error.

**Comment by eywalker** [comment]
Oops, that's my typo. Updated the example.

**Comment by dimitri-yatsenko** [comment]
Wasn't this fixed with #342?  Do you have the latest version?

**Comment by eywalker** [comment]
You are absolutely right about this - there was an issue in my local env such that it wasn't pointing to the latest DataJoint version - I created and tested this on an environment with the latest release and sure enough this issue was fixed. My bad :stuck_out_tongue:
